python/ray/worker.py

```diff
@@ -1290,10 +1290,17 @@ def connect(node,
         script_directory = os.path.abspath(os.path.dirname(sys.argv[0]))
         worker.run_function_on_all_workers(
             lambda worker_info: sys.path.insert(1, script_directory))
-        if not job_config.client_job and job_config.get_runtime_env_uris():
+        if not job_config.get_runtime_env_uris():
             current_directory = os.path.abspath(os.path.curdir)
-            worker.run_function_on_all_workers(
-                lambda worker_info: sys.path.insert(1, current_directory))
+
+            def setup_dir():
+                # Here "" means search cwd
+                if "" not in sys.path:
+                    sys.path.insert(0, "")
+                # Change cwd so that it can open local file
+                os.chdir(current_directory)
+
+            worker.run_function_on_all_workers(lambda worker_info: setup_dir())
```

Comment:
Hmmm why do we need to use `run_function_on_all_workers` here? Is there any reason we can't just do this in the setup code in `default_worker.py`?

Reply:
Oh, I'm just following the existing code. [This](https://github.com/ray-project/ray/blob/d4c20c970b15f32d6bef7dadd2ad9e1db9b34132/python/ray/worker.py#L1285-L1288) is before my change.

I did one change [before](https://github.com/ray-project/ray/pull/15111/files) to add it to sys.path. But the thing here is that, we actually need to change dir to make the worker able to check the file. 


I feel these workers should be the ones already there. And when the second init called and they somehow got reused. But since cwd is still the old one, although it's able to access the code (we add it to sys.path), it can't access the file (cwd is still the old one).

Reply:
This is very fishy, I don't think we should be using that function for runtime env...

Reply:
@ericl  it's not for runtime environment actually. If you check the test case I added, it's in `test_basic.py`

So if we'd like to let the user open the file in a different directory, they can't do that.

Reply:
The reason is that the existing worker is using the previous job's working dir which prevent it reading file from the existing job's working dir. So we must change the working dir to fix this. 

- Working dir is to read files
- sys.path is to search library 

So we have to have these two to make it complete.